core/src/main/scala/spire/math/Complex.scala

```diff
@@ -119,9 +119,9 @@ final case class Complex[@sp(Float, Double) T](real: T, imag: T)
   def asTuple: (T, T) = (real, imag)
   def asPolarTuple(implicit f: Field[T], n: NRoot[T], s: Signed[T], t: Trig[T]): (T, T) = (abs, arg)
 
-  def isZero(implicit s: Signed[T]): Boolean = real.isSignZero && imag.isSignZero
-  def isImaginary(implicit s: Signed[T]): Boolean = real.isSignZero
-  def isReal(implicit s: Signed[T]): Boolean = imag.isSignZero
+  def isZero(implicit s: Signed[T]): Boolean = real.isSignZero() && imag.isSignZero()
```

Comment:
Many of the changed files is because scala 3 requires the empty parens. I'd like in the near future remove all of them but not in this PR

Reply:
That's a really great point about compiler flags. That would be a much better solution.

Reply:
They produce warnings. My preference would be to remove the parenthesis everywhere which I'll do in a separate PR.
I know it is an API change but IMHO it is worth doing.

Reply:
I agree, so since this is all going to be change in a later PR anyway can we keep the warnings here and just reduce the size of the diff?

Reply:
I can include my removal of parenthesis commit on this PR and will reduce the diff size